src/assets/Generator.Shared/HttpPipelineExtensions.cs

```diff
@@ -0,0 +1,112 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#nullable enable
+
+using System;
+using System.Threading;
+using System.Threading.Tasks;
+using Azure.Core.Pipeline;
+
+namespace Azure.Core
+{
+    internal static class HttpPipelineExtensions
+    {
+#if EXPERIMENTAL
+        public static async ValueTask<Response> ProcessMessageAsync(this HttpPipeline pipeline, HttpMessage message, ClientDiagnostics clientDiagnostics, RequestOptions? requestOptions, CancellationToken cancellationToken = default)
```

Comment:
This method incapsulates some of the code that has been previously generated, as well as new `if (statusOption == ResponseStatusOption.NoThrow || !message.ResponseClassifier.IsErrorResponse(message))` piece